bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBluetoothDevice.java

```diff
@@ -0,0 +1,474 @@
+                        BluetoothCompletionStatus.SUCCESS);
+
+            } catch (DBusException e) {
+                logger.debug("Exception occurred when trying to write characteristic '{}': {}",
+                        characteristic.getUuid(), e.getMessage());
+                notifyListeners(BluetoothEventType.CHARACTERISTIC_WRITE_COMPLETE, characteristic,
+                        BluetoothCompletionStatus.ERROR);
+            }
+        });
+        return true;
+    }
+
+    @Override
+    public void onDBusBlueZEvent(BlueZEvent event) {
+        logger.debug("Unsupported event: {}", event);
+    }
+
+    @Override
+    public void onServicesResolved(ServicesResolvedEvent event) {
+        if (event.isResolved()) {
+            notifyListeners(BluetoothEventType.SERVICES_DISCOVERED);
+        }
+    }
+
+    @Override
+    public void onNameUpdate(NameEvent event) {
+        BluetoothScanNotification notification = new BluetoothScanNotification();
+        notification.setDeviceName(event.getName());
+        notifyListeners(BluetoothEventType.SCAN_RECORD, notification);
+    }
+
+    @Override
+    public void onManufacturerDataUpdate(ManufacturerDataEvent event) {
+        for (Map.Entry<Short, byte[]> entry : event.getData().entrySet()) {
+            BluetoothScanNotification notification = new BluetoothScanNotification();
+            byte[] data = new byte[entry.getValue().length + 2];
+            data[0] = (byte) (entry.getKey() & 0xFF);
+            data[1] = (byte) (entry.getKey() >>> 8);
+
+            System.arraycopy(entry.getValue(), 0, data, 2, entry.getValue().length);
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("Received manufacturer data for '{}': {}", address, HexUtils.bytesToHex(data, " "));
+            }
+
+            notification.setManufacturerData(data);
+            notifyListeners(BluetoothEventType.SCAN_RECORD, notification);
+        }
+    }
+
+    @Override
+    public void onTxPowerUpdate(TXPowerEvent event) {
+        this.txPower = (int) event.getTxPower();
+    }
+
+    @Override
+    public void onCharacteristicNotify(CharacteristicUpdateEvent event) {
+        // Here it is a bit special - as the event is linked to the DBUS path, not characteristic UUID.
+        // So we need to find the characteristic by its DBUS path.
+        BluetoothGattCharacteristic characteristic = getDBusBlueZCharacteristicByDBusPath(event.getDbusPath());
+        if (characteristic == null) {
+            logger.info("Received a notification for a characteristic not found on device.");
+            return;
+        }
+        BluetoothCharacteristic c = getCharacteristic(UUID.fromString(characteristic.getUuid()));
+        if (c != null) {
+            c.setValue(event.getData());
+            notifyListeners(BluetoothEventType.CHARACTERISTIC_UPDATED, c, BluetoothCompletionStatus.SUCCESS);
+            logger.debug("Has notified that a characteristic has been updated");
```

Comment:
```suggestion
            logger.debug("Notified that a characteristic has been updated");
```

Reply:
Actually, this log isn't really needed anyway since I'll get a log from the BlueZPropertyChangedHandler anyway.
Removing.